source/agora/common/Config.d

```diff
@@ -64,6 +64,14 @@ public struct CommandLine
     }
 }
 
+/// Thrown when there is a configuration error
+class ConfigException : Exception
+{
+    this(string msg, string file = __FILE__, size_t line = __LINE__) {
+        super(msg, file, line);
+    }
+}
+
```

Comment:
None of the changes in this file seems to be relevant. It seems you are introducing `ConfigException` only to transform it into `Exception` soon after. I can't see any special handling of `ConfigException` vs `Exception`.

Reply:
I transform Exception to ConfigException and catch ConfigException from the main

Reply:
But... Why not just catch `Exception` as before ?

Reply:
Because that would catch the Exception thrown from the file locker, and also the current catch block in the main say the exception is related to 'config' issues

Reply:
if you would prefer I could do this:
1. leave the ConfigException as it is in the main
2. throw a specific LockException and catch that in the main too

every other exception should propage through main(although there is none current as the runTask doesn't leak Exceptions)

Reply:
Well, there's a few ways around it:
- We expose `lock` and `lockThrow`. Why ? We could just have a boolean and avoid the `throw`ing part;
- We don't need to modify the config, we can just add a specific exception to `lock` and catch it first;
- Change the message in a way that does not make assumption, *or* break the `try` block up;

Actually the later sounds like the best idea. Let me give it a quick try.

Reply:
"We expose lock and lockThrow. Why ?
we have 3 public functions
lockTry, lockBlock, lockThrow we don't expose the lock function, I think it is the lockTry that return a boolean. 

"We don't need to modify the config, we can just add a specific exception to lock and catch it first;"
I think it is a bit messy, it is not guaranteed that every Exception is a ConfigException

How about my idea of having specific LockException and LockConfig: we could just create a simple mixin and use to define those exception classes like:
1. mixin CustomException!(LockException);
2. mixin CustomException!(ConfigException);

Reply:
should I do 
'Change the message in a way that does not make assumption, or break the try block up;'?

Reply:
> Change the message in a way that does not make assumption, or break the try block up;

Did the latter [here](https://github.com/bpfkorea/agora/pull/1153).

> I think it is a bit messy, it is not guaranteed that every Exception is a ConfigException(in the future)

Agreed, that's why IMO reducing the scope to just `parseConfigFile` would be better.